### DISCUSSION TITLE
PHP CLI container without services dependencies

### DISCUSSION THREAD

**Discussion by marciomaciel** [wardenenv/warden#808] [Polls]
Hi everyone,

In my local development environment using Warden, I created a specific container for running PhpStorm’s quality tools inspections. The issue I ran into is that services like php-fpm and php-debug depend on others, like MySQL. So, when running the inspection in PhpStorm, it needs MySQL to be up, which leads to unnecessary services being started. This check to bring up MySQL adds extra time to the process and creates a delay when running code sniffer or using the remote interpreter as an external formatter.

Locally, I solved this by creating a specific service called **php-inspection** without any dependencies, but I only did this for the Magento 2 environment. I want to know if the community sees value in this implementation. If so, I can contribute by adding support for this inspection service across all PHP environments.

Would you be interested in this? If there’s enough interest, I’ll create a PR with the necessary adjustments and documentation update

**Comment by navarr** [comment]
I'm confused by the scenario where you use Warden for PHP inspections against Magento but don't need other Magento required services like MySQL.

Could you explain your use case a little more please?

**Reply by bap14** [reply]
@navarr I'm guessing ~he's~ they're doing static analysis or code compliance with PHP Code Sniffer. Neither of which require the full suite of Magento services to scan the code; however, the tools need PHP to run. That's just a guess though.

What I'm not understanding is presumably you're using those tools to check code you just wrote and presumably you're functionality testing it in the browser before then running the analysis tools on them.

This is just a guess, but I'm wondering if the perceived slow down from the database is more so in Mutagen syncing the files and having to wait for that process to finish before they can run `warden shell` and then run their analysis tool(s).
